Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperations/TestOperationResult.swift

```diff
@@ -20,27 +20,31 @@ enum TestOperationResult: Decodable, Equatable, Matchable {
     /// Result of test operations that are expected to return an error (e.g. `CommandError`, `WriteError`).
     case error(ErrorResult)
 
-    public init?(from doc: Document?) {
-        guard let doc = doc else {
+    public init<T: Codable>(cursor: () throws -> MongoCursor<T>) throws {
```

Comment:
these were all updated to allow for the trailing closure syntax to magically convert things to a `TestOperationResult`.

e.g.
```swift
try TestOperationResult {
    try MongoClient().db("test").collection("coll").find(["x" : 1])
}
```
I'm not certain the "magic" syntax is better than being explicit though, so let me know what you think.